indexer/run.sh

```diff
@@ -33,23 +33,88 @@ fetch --nohooks chromium
 
 cd src
 
+mkdir -p out/Build
+export BUILD_DIR=$(readlink -f out/Build)
+
+DATE=$(date -u +%Y%m%d)
+
+# $1 is the directory where the build will live.
+# $2 is the platform name.
+# $3 is the date that will be put into the index filename.
+index() {
+  gclient sync -D
+
+  gclient runhooks
+
+  gn gen $1
+
+  ninja -C $1 -t targets all | grep -i '^gen/' | grep -E "\.(cpp|h|inc|cc)\:" | cut -d':' -f1 | xargs autoninja -C $1
+
+  tools/clang/scripts/generate_compdb.py -p $1 > compile_commands.json
+
+  $CLANGD_INDEXER --executor=all-TUs compile_commands.json > /chrome-$2.idx
+
+  7z a chrome-index-$2-$3.zip /chrome-$2.idx
+
+  # Clean up the build directory afterwards.
+  rm -rf $1
+}
+
+# --- Linux ---
+
+PLATFORM="linux"
+
+echo "Indexing for $PLATFORM"
+
 # Remove snapcraft from dependency list: installing it is not feasible inside
 # Docker.
 sed -i '/if package_exists snapcraft/,/fi/d' ./build/install-build-deps.sh
 ./build/install-build-deps.sh
 
-gclient runhooks
+index $BUILD_DIR $PLATFORM $DATE
+
+# --- Android ---
+
+PLATFORM="android"
+
+echo "Indexing for $PLATFORM"
+
+echo "target_os = [ '$PLATFORM' ]" >> ../.gclient
```

Comment:
i thought we were going to put all the `target_os`s we are interested in gclient and fetch all of them at once, then perform build/indexing with different `gn args`.

That way we can also guarantee different build configs won't see different versions of the sources and also get rid of the latency we spend for fetching sources multiple times.